File: Python/38_count_and_say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        res = "1"
        while n > 1:
            i, j, tmp = 0, 0, ""
            while i < len(res):
                j = i + 1
                count = 1
                while j < len(res):
                    if res[j] == res[i]:
                        count += 1
                    else:
                        break
                    j += 1
                tmp += str(res[i]) + str(count)
                i = j
            res = tmp
            n -= 1
        return res

s = Solution()
res = s.countAndSay(3)
print(res)

File: leetcode/archived/golang/38_count_and_say.go
func countAndSay(n int) string {
	res := "1"
	for n > 1 {
		var i, count int = 0, 1
		var tmp string = ""
		for i < len(res){
			j := i + 1
			count = 1
			for j  <  len(res){
				if res[i] == res[j] {
					count += 1
				}else {
					break
				}
				j += 1
			}
			tmp += strconv.Itoa(count) +string(res[i])
			i = j
		}
		res = tmp
		n -= 1
	}
	return res
}